libcrux-ml-dsa/src/simd/avx2.rs

```diff
@@ -137,9 +137,13 @@ impl Operations for AVX2SIMDUnit {
         }
         let result = ntt::ntt(re);
 
-        core::array::from_fn(|i| Self {
-            coefficients: result[i],
-        })
+        let mut out = [vector_type::ZERO(); SIMD_UNITS_IN_RING_ELEMENT];
+        for i in 0..result.len() {
+            out[i] = Self {
+                coefficients: result[i],
+            };
+        }
+        out
```

Comment:
Are there some patterns of `from_fn` we should not use?

Reply:
This used to work. I'm not sure what broke. But from_fn has always been flaky. And all this code should go away soon. I haven't figured out a pattern yet.